### ISSUE TITLE
Keybind configuration

### ISSUE THREAD

**Issue by Xithrius** [Xithrius/twitch-tui#362]
Some users may want different things, so let customization be a thing.

**Comment by Xithrius** [comment]
Thank you!

**Comment by Xithrius** [comment]
Keybinds and CLI args have been updated, you might want to rebase.

**Comment by marcoradocchia** [comment]
Sure, I'll do. Thanks for the warning. I've been a little busy, but I'm slowly working on it. I'll update you soon on the progress.

**Comment by Xithrius** [comment]
Hey @marcoradocchia, would you still like to work on this?

**Comment by marcoradocchia** [comment]
Hey @! I'm sorry to not have produced a PR yet. I have made progress, but it still does need some work. Would still like to contribute though, if you want me to. I might have some time tonight to work on it and update you on the status.

**Comment by Xithrius** [comment]
I have no problem with you working on this. However, I took a look at your branch, and you're around 100 commits behind main. Hopefully there aren't that many conflicts, but if there are, might be a good idea to start over.

**Comment by marcoradocchia** [comment]
It's fine, I'm working on it right now.
I was thinking about providing the user three TOML tables, one for each mode, like the following:
```toml
[keybind.normal]
# Normal keybinds...

[keybind.insert]
# Insert keybinds...

[keybind.dashboard]
# Dashboard keybinds...
```
What do you think about it?
Also, is there any requirement for the configuration format? Should it be a `+` separated a string containing key modifiers and a single keycode, maybe like `"ALT + CTRL + p"`, or something else? I would like your opinion about this :)

**Comment by Xithrius** [comment]
Yeah, I think those tables will work. We can always change them in the future, of course. Along with that, I don't see a problem with the `+` separated keybinds.

**Comment by marcoradocchia** [comment]
I have a `keybind` module, it does serialize/deserialize keybinds to/from TOML.
I've been writing some unit tests for it and they pass. It all seems to work as expected.
The three `keybind.normal`, `keybind.insert` and `keybind.dashboard` tables are now also available for keybind configuration.

I'm going ahead and work on integration now :)

**Comment by marcoradocchia** [comment]
Still working on it... I'm sorry for the waiting, in this period I should have more time.

**Comment by CharlesBuell** [comment]
No problem! Thank you for the continued updates.

**Comment by Xithrius** [comment]
No problem! Thank you for the continued updates.

**Comment by marcoradocchia** [comment]
I'm now trying to update the help widget to dynamically display the configured keybinds. One important question about it: do you have any plans about implementing hot reloading of the configuration file?

**Comment by Xithrius** [comment]
I don't have any plans at the moment to do so.

**Comment by marcoradocchia** [comment]
May the user be able to configure multiple keybinds for the same action?
For reference, see: https://github.com/Xithrius/twitch-tui/blob/2cbb4b8f847494504c0f4e157921070d72dac3a8/src/ui/components/utils/input_widget.rs#L179-L181

**Comment by Xithrius** [comment]
Yes

**Comment by marcoradocchia** [comment]
Nice, thanks

**Comment by Xithrius** [comment]
@marcoradocchia Hey, checking up. How's progress?
